core/arch/x86_64/kernel/thread.c

```diff
@@ -728,14 +728,9 @@ void thread_state_save(vaddr_t sp, uint32_t client)
 
 	assert(threads[ct].state == THREAD_STATE_ACTIVE);
 
-	if (client == TEE_LOGIN_TRUSTED_APP) {
```

Comment:
You may need to remove client parameter as well.

Reply:
done